drivers/drivers.go

```diff
@@ -0,0 +1,11 @@
+package drivers
+
+// The Driver interface is the set of functions required to start, manage, and verify a Kubernetes cluster. Inspired by the steps used in kube-up.sh, with extensions for management and scaling of clusters.
+type Driver interface {
+	GenerateCerts()
+	GetTokens()
+	ProvisionMaster()
+	ConfigureMaster()
+	ProvisionNode()
```

Comment:
Reference for you:

https://github.com/kubernetes/kubernetes/blob/master/cluster/kube-up.sh
Vagrant/VBox implementation: https://github.com/kubernetes/kubernetes/blob/master/cluster/vagrant/util.sh
docker-machine, especially drivers and providers: https://github.com/docker/machine